internal/controllers/core/cmd/controller.go

```diff
@@ -138,10 +138,11 @@ func (c *Controller) fileWatches(ctx context.Context, cmd *v1alpha1.Cmd) (map[st
 }
 
 // Fetch the last time a start was requested from this target's dependencies.
-func (c *Controller) lastStartEventTime(startOn *StartOnSpec, buttons map[string]*v1alpha1.UIButton) time.Time {
-	cur := time.Time{}
+func (c *Controller) lastStartEvent(startOn *StartOnSpec, buttons map[string]*v1alpha1.UIButton) (time.Time, []v1alpha1.UIInputStatus) {
+	latestTime := time.Time{}
+	var latestButton *v1alpha1.UIButton
```

Comment:
UIButtons also have a DefaultValue field. how does that play out here? i kind of expected the DefaultValue would be injected into the env, even if the button wasn't pressed. though maybe I'm misunderstanding the intended behavior of DefaultValue

Reply:
Whoops, I typed an answer to this but I guess I didn't submit?

The `UITextInput`'s `DefaultValue` is passed as the `TextField`'s `defaultValue`. When the user clicks the button, the frontend updates the `UITextInput`'s status with the `TextField`'s current value, which will already be making use of the default as appropriate.